config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotableEntity.java

```diff
@@ -26,43 +22,71 @@
      */
     String getType();
 
+    /**
+     * This method returns all the Annotation applied to the entity.
+     *
+     * @return Set of all the annotations
+     */
+    Set<Annotation> getAnnotations();
+
     /**
      * Creates AnnotatedEntity object by scanning all the annotations and gathering all the information required to
      * generate the bundle and its metadata.
      *
-     * @param annotations
      * @return AnnotatedEntity
      */
-    default AnnotatedEntity<GatewayEntity> createAnnotatedEntity(final Set<Annotation> annotations) {
+    default AnnotatedEntity<GatewayEntity> createAnnotatedEntity() {
+        final Set<Annotation> annotations = getAnnotations();
         if (annotations != null) {
             AnnotatedEntity<GatewayEntity> annotatedEntity = new AnnotatedEntity(this);
             annotations.forEach(annotation -> {
-                switch (annotation.getType()) {
-                    case ANNOTATION_TYPE_BUNDLE:
-                        annotatedEntity.setTags(annotation.getTags());
-                        annotatedEntity.setBundle(true);
-                        annotatedEntity.setEntityType(EntityTypes.ENCAPSULATED_ASSERTION_TYPE);
-                        String annotatedBundleName = annotation.getName();
-                        annotatedEntity.setBundleName(annotatedBundleName);
-                        String description = annotation.getDescription();
-                        annotatedEntity.setDescription(description);
-                        break;
-                    case ANNOTATION_TYPE_REUSABLE_ENTITY:
-                        annotatedEntity.setReusableEntity(true);
-                        break;
-                    case ANNOTATION_TYPE_REDEPLOYABLE:
-                        annotatedEntity.setRedeployable(true);
-                        break;
-                    case ANNOTATION_TYPE_EXCLUDE:
-                        annotatedEntity.setExclude(true);
-                        break;
-                    default:
-                        break;
+                if (ANNOTATION_TYPE_BUNDLE.equalsIgnoreCase(annotation.getType())) {
+                    annotatedEntity.setTags(annotation.getTags());
+                    annotatedEntity.setEntityType(EntityTypes.ENCAPSULATED_ASSERTION_TYPE);
+                    String annotatedBundleName = annotation.getName();
+                    annotatedEntity.setBundleName(annotatedBundleName);
+                    String description = annotation.getDescription();
+                    annotatedEntity.setDescription(description);
                 }
             });
             return annotatedEntity;
         }
         return null;
     }
 
+    /**
+     * Returns TRUE if "@bundle" annotation is added to the entity
+     *
+     * @return TRUE if "@bundle" annotation is added
+     */
+    default boolean isBundle() {
+        return getAnnotations() != null && getAnnotations().contains(new Annotation(ANNOTATION_TYPE_BUNDLE));
```

Comment:
why do we have to refer annotations every time to find if its bundle or not
AnnotatedEntity should have this

Reply:
This is required to make sure we always refer to the same source of truth which is the annotations added to the entity. And since `getAnnotations()` returns a `Set`, the `contains()` operation has constant time complexity so its almost same as reading the variable value.

Reply:
Can you avoid creating new Annotation object while calling these methods?

Reply:
@graju256 fixed it.